geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExpireDUnitTest.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.geode.redis.internal.executor.TTLExecutor;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+
+@Ignore("GEODE-8058: this test needs to pass to have feature parity with native redis")
+public class ExpireDUnitTest {
+
+  @ClassRule
+  public static ClusterStartupRule clusterStartUp = new ClusterStartupRule(4);
+
+  static final String LOCAL_HOST = "127.0.0.1";
+  static int[] availablePorts;
+  private static final int JEDIS_TIMEOUT =
+      Math.toIntExact(GeodeAwaitility.getTimeout().toMillis());
+  static Jedis jedis1;
+  static Jedis jedis2;
+  static Jedis jedis3;
+
+  static Properties locatorProperties;
+  static Properties serverProperties1;
+  static Properties serverProperties2;
+  static Properties serverProperties3;
+
+  static MemberVM locator;
+  static MemberVM server1;
+  static MemberVM server2;
+  static MemberVM server3;
+
+  @BeforeClass
+  public static void classSetup() {
+
+    availablePorts = AvailablePortHelper.getRandomAvailableTCPPorts(3);
+
+    locatorProperties = new Properties();
+    serverProperties1 = new Properties();
+    serverProperties2 = new Properties();
+    serverProperties3 = new Properties();
+
+    locatorProperties.setProperty(MAX_WAIT_TIME_RECONNECT, "15000");
+
+    serverProperties1.setProperty(REDIS_PORT, Integer.toString(availablePorts[0]));
+    serverProperties1.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    serverProperties2.setProperty(REDIS_PORT, Integer.toString(availablePorts[1]));
+    serverProperties2.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    serverProperties3.setProperty(REDIS_PORT, Integer.toString(availablePorts[2]));
+    serverProperties3.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    locator = clusterStartUp.startLocatorVM(0, locatorProperties);
+    server1 = clusterStartUp.startServerVM(1, serverProperties1, locator.getPort());
+    server2 = clusterStartUp.startServerVM(2, serverProperties2, locator.getPort());
+    server3 = clusterStartUp.startServerVM(3, serverProperties3, locator.getPort());
+
+    jedis1 = new Jedis(LOCAL_HOST, availablePorts[0], JEDIS_TIMEOUT);
+    jedis2 = new Jedis(LOCAL_HOST, availablePorts[1], JEDIS_TIMEOUT);
+    jedis3 = new Jedis(LOCAL_HOST, availablePorts[2], JEDIS_TIMEOUT);
+  }
+
+  @After
+  public void testCleanUp() {
+    jedis1.flushAll();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+    jedis3.disconnect();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void expireOnOneServer_shouldPropagateToAllServers() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+
+    assertThat(jedis2.ttl(key)).isGreaterThan(0);
+  }
+
+  @Test
+  public void expireOnOneServer_shouldResultInKeyRemovalFromOtherServer() {
```

Comment:
this is the same as the one above except for the extra.  Can we combine them, or have mutually exclusive asserts?

Reply:
They're a little different since one checks to make sure the key is removed after expiration, but we could remove the duplicate assert.